messages/teleporter/message_manager.go

```diff
@@ -87,13 +109,59 @@ func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageI
 	if !ok {
 		return false, fmt.Errorf("relayer not configured to deliver to destination. destinationChainID=%s", destinationChainID.String())
 	}
-	if !destinationClient.Allowed(destinationChainID, teleporterMessage.AllowedRelayerAddresses) {
+	senderAddress := destinationClient.SenderAddress()
+	if !isAllowedRelayer(teleporterMessage.AllowedRelayerAddresses, senderAddress) {
+		m.logger.Info("Relayer EOA not allowed to deliver this message.")
+		return false, nil
+	}
+	if !isDestination(destinationChainID, destinationClient.DestinationChainID()) {
 		m.logger.Info(
-			"Relayer not allowed to deliver to chain.",
+			"Destination chain ID for message not supported by relayer.",
 			zap.String("destinationChainID", destinationChainID.String()),
 		)
 		return false, nil
 	}
+
+	// Check if the message has already been delivered to the destination chain
```

Comment:
Could we move the section below to a `messageAlreadyDelivered` helper? (or whatever name you think is best)

Reply:
Sure thing, changed